src/webapi/rendering_context.rs

```diff
@@ -30,6 +46,201 @@ impl RenderingContext for CanvasRenderingContext2d {
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/clip)
+    pub fn clip(&self, fill_rule: Option<FillRule>) {
+        if let Some(fill_rule) = fill_rule {
+            let fill_rule_str;
+            match fill_rule {
+                FillRule::NonZero => {
+                    fill_rule_str = "nonzero";
+                }
+
+                FillRule::EvenOdd => {
+                    fill_rule_str = "evenodd";
+                }
+            }
+            js! { @(no_return)
+                @{&self.0}.clip(@{fill_rule_str});
+            }    
+        }
+        else {
+            js! { @(no_return)
+                @{&self.0}.clip();
+            }
+        }
+    }
+
+    /// Causes the point of the pen to move back to the start of the current sub-path. It tries 
+    /// to add a straight line (but does not actually draw it) from the current point to the start. 
+    /// If the shape has already been closed or has only one point, this function does nothing.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/closePath)
+    pub fn close_path(&self) {
+        js! { @(no_return)
+            @{&self.0}.closePath();
+        }
+    }
+
+    /// Creates a gradient along the line given by the coordinates represented by the parameters.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createLinearGradient)
+    pub fn create_linear_gradient(&self, x0: f64, y0: f64, x1: f64, y1: f64) {
+        //TODO: returns linear gradient
+        js! { @(no_return)
+            @{&self.0}.createLinearGradient(@{x0}, @{y0}, @{x1}, @{y1});
+        }
+    }
+
+    /// Creates a pattern using the specified image (a CanvasImageSource). It repeats the source in 
+    /// the directions specified by the repetition argument. This method returns a CanvasPattern.
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern)
+    pub fn create_pattern_image(&self, image: ImageElement, repitition: Option<Repitition>) {
+        let repitition_string;
+        if let Some(repitition) = repitition {
+            match repitition {
+                Repitition::Repeat => {
+                    repitition_string = "repeat";
+                }
+
+                Repitition::RepeatX => {
+                    repitition_string = "repeat-x";
+                }
+
+                Repitition::RepeatY => {
+                    repitition_string = "repeat-y";
+                }
+
+                Repitition::NoRepeat => {
+                    repitition_string = "no-repeat";  
+                }
+            }
+        }
+        else {
+            repitition_string = "repeat";
+        }
```

Comment:
You can do this with a single `match` statement. (In this and other methods.)

```
let repetition = match repitition {
    Some( Repetition::Repeat ) | None => "repeat",
    Some( Repetition::RepeatX ) => "repeat-x",
};
```

Also, a typo - `repitition` -> `repetition`. (:

Reply:
Thanks! That's a lot cleaner :)